rfc/mentorship-rfc.md

```diff
@@ -0,0 +1,89 @@
+# _Spinnaker Mentorship Program Proposal_
+
+| | |
+|-|-|
+| **Status**     | _**Proposed**, Accepted, Implemented, Obsolete_ |
+| **RFC #**      | _Update after PR has been made_ |
+| **Author(s)**  | _Emily Burns (`@emjburns`)_ |
+| **SIG / WG**   | _?_ |
+
+## Overview
+
+This proposal describes an idea for a mentorship program for the Spinnaker OSS community that Ethan Rogers (`@ethanfrogers`), Rob Zienert (`@robzienert`), and I came up with.
+
+
+### Goals and Non-Goals
+
+The goal of this program is to create a more visible and structured way for new contributers to get the feedback they need to contribute successfully.
+
+A secondary goal is to creating more Approvers in the community and to encourage existing Approvers to provide more detailed PR review. 
+
+
+## Motivation and Rationale
+
+
+As a new (or prospective) contributor it's hard to break into a community where you have few connections. It's hard to understand how to add functionality or make improvements to Spinnaker. It's complicated to figure out who has the expertise to review your PR or provide design guidance.
+
+As a maintainer, lots of time is spent providing feedback and PR review. That doesn't always lead to more quality PRs.
+
+I'd like to create a more visible path to becoming an Approver (and the oportunity for help along the way).
+
+
+## Design
+
+### Find Mentors
+
+To initially find mentors we would pick from current approvers in the community. We’d like to select contributors who are already producing quality PRs and who know a decent amount about Spinnaker architecture. Mentors must be willing to give prompt PR feedback, answer mentee questions on Slack, and generally guide a mentee for several months.
+
+### Align Mentors
+
+Mentors will meet together and set expectations on how to interact with mentees so that mentees have a consistent experience. 
+
+### Find Mentees
+
+Mentees will be found through self-nomination. We will put out a google form, and anyone in the community could sign up to be part of the mentorship program. From there, we will pick the same number of mentees as we have mentors. We want these mentees to be spread out between as many different companies as possible.
+
+This google form will contain some information about what areas of interest a mentee has and what area of the code they’re planning to continue working in to help us match people up.
+
+### Do Mentorship
+
+The mentorship program would last for 3-6 months (hard stop at 6 months). This timeline is somewhat flexible to account for light/heavy quarters of work. It’s really up to the judgment of the pair on whether it’s working or not. 
```

Comment:
A few questions that come to mind - 
* Does a mentor immediately get a new mentee as soon as the 3-6 months is up or do they get a break? Up to them?
* If this program proves successeful, should we start mentorship cycles each quarter or every 6 months?

Reply:
* I think it's up to them if they want to participate again or take a break.
* I think it would be good to start every quarter, but I'd like to evaluate it based on need.